### ISSUE TITLE
Cannot schedule a coupon via CRUD

### ISSUE THREAD

**Issue by seldimi** [woocommerce/woocommerce#26329]
Hello.
I'm trying to schedule a coupon via CRUD.
My code is simple.

```
$coupon_code="test_schedule";
$amount="10";

$discount_type = 'fixed_cart';
$coupon = new WC_Coupon();
$coupon->set_code($coupon_code);
$coupon->set_discount_type($discount_type);
$coupon->set_amount($amount);
$coupon->set_date_expires(date('Y-m-d H:i:s', strtotime('+4 months')));
$coupon->set_date_created(date( 'Y-m-d H:i:s', strtotime('+1 day') ));
$coupon->set_date_modified(date( 'Y-m-d H:i:s', strtotime('+1 day') ));

$coupon->set_usage_count(0);
$coupon->set_individual_use(true);
$coupon->set_usage_limit(1);
$coupon->set_exclude_sale_items(true);
$coupon->set_free_shipping(false);
$coupon->set_email_restrictions(array("test@domain.com"));
$coupon->save();
```

The coupon is created with current date, not future date.

If I do same action from WooCommmerce > Coupons, I can edit the creation/edit date so I can schedule it.

**Comment by tammullen** [comment]
Hi @seldimi 

While our goal is to address all the issues reported in this repository, GitHub should be treated as a place to report confirmed bugs only.
​
The type of issue you submitted looks like a custom code related question. Due to the number of bug reports being submitted, we are prioritizing work on them and are not able to attend to custom code questions. 
​
The best place to get help further is in the [WooCommerce Community Slack](https://woocommerce.com/community-slack/) in the #developers channel. 
​
If you confirm the bug, please open a new issue and provide us with clear steps to reproduce it.
​
I am closing this issue for now. If I missed something, please leave a comment and we will take a second look.
Label:

**Comment by seldimi** [comment]
I'm sorry but woocommerce CRUD is not custom work but part of Woocommerce. On this repository you will see the WC_Coupon class.

https://docs.woocommerce.com/document/developing-using-woocommerce-crud-objects/

https://github.com/woocommerce/woocommerce/wiki/Coupon-Data

**Comment by tammullen** [comment]
Hi @seldimi 

If you confirm there is a potential bug please fill out the details of the bug report template that we ask for. Without these details, we can’t fully evaluate this issue. Please provide us with the information requested so we could take a look further.
​
**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed without action.
​
**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
​
**Screenshots**
If applicable, add screenshots to help explain your problem.
​
**Expected behavior**
A clear and concise description of what you expected to happen.
​
**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.
​
**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

**Comment by seldimi** [comment]
OK will try to give a more detailed description  Tommorow. Thanks.

**Comment by claudiosanches** [comment]
@seldimi I managed to reproduce this issue, there's a fix for it already that you can test in #26387

**Comment by seldimi** [comment]
Thank you. 😃 looking forward to make it live. Sorry I couldn't report in more detail.
